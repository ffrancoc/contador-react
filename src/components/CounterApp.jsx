import { useState } from 'react'
import PropTypes from 'prop-types'
import './CounterApp.css'

function CounterApp({value}) {

  const [contador, setContador] = useState(value)

  const incrementHandle = () => {
    setContador(contador + 1)
  }

  const decrementHandle = () => {
    if ( contador > 0 ){
      setContador(contador - 1)
    }
  }

  const resetHandle = () => {
    setContador(value)
  }

  return (
    <div className='app-container'>
      <h1>Simple contador en React</h1>
      <h2 aria-label='contador'>{ contador }</h2>
      <div>
        <button data-testid='btnincrement' onClick={ incrementHandle }>+</button>
        <button data-testid='btndecrement' onClick={ decrementHandle }>-</button>
        <button data-testid='btnreset' onClick={ resetHandle }>Reset</button>
      </div>
    </div>
  )
}

CounterApp.propTypes = {
  value: PropTypes.number.isRequired
}

CounterApp.defaultProps = {
  value: 10
}

export default CounterApp