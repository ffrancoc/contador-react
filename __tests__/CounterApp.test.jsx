import { fireEvent, render, screen } from "@testing-library/react";
import CounterApp from "../src/components/CounterApp";

describe('Pruebas sobre el elemento <App />', () => {
    const initialValue = 10

    test('deberia hacer match con el snapshot', () => {
        const { container } = render(<CounterApp />)
        expect(container).toMatchSnapshot()
    });

    test('deberia validar que el valor inicial del contador sea 10', () => {
        render(<CounterApp value={initialValue}/>)        
        expect(screen.getByRole('heading', {level: 2}).innerHTML).toContain(`${initialValue}`)
    });
    
    test('deberia incrementar el contador en +1', () => {
        render(<CounterApp value={initialValue}/>)        
        fireEvent.click(screen.getByTestId('btnincrement'))    
        expect(screen.getByLabelText('contador').innerHTML).toContain(`${initialValue + 1}`)
    });
    
    test('deberia decrementar el contador en -1', () => {
        render(<CounterApp value={initialValue}/>)        
        fireEvent.click(screen.getByTestId('btndecrement'))    
        expect(screen.getByLabelText('contador').innerHTML).toContain(`${initialValue - 1}`)
    });
    
    test('deberia reiniciar el contador al valor por defecto', () => {
        render(<CounterApp value={initialValue}/>)        
        fireEvent.click(screen.getByTestId('btnreset'))    
        expect(screen.getByLabelText('contador').innerHTML).toContain(`${initialValue}`)        
    });
})