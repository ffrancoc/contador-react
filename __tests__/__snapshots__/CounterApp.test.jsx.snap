// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre el elemento <App /> deberia hacer match con el snapshot 1`] = `
<div>
  <div
    class="app-container"
  >
    <h1>
      Simple contador en React
    </h1>
    <h2
      aria-label="contador"
    >
      10
    </h2>
    <div>
      <button
        data-testid="btnincrement"
      >
        +
      </button>
      <button
        data-testid="btndecrement"
      >
        -
      </button>
      <button
        data-testid="btnreset"
      >
        Reset
      </button>
    </div>
  </div>
</div>
`;
